### DISCUSSION TITLE
Azure - Storage Account - "HIGH: Network rules do not allow bypass for Microsoft Services."

### DISCUSSION THREAD

**Discussion by AlexcFrench** [aquasecurity/trivy#8055] [False Detection]
### IDs

Not Applicable

### Description

We have multiple storage accounts that do not allow Microsoft Services to bypass the network ACLS. 

This is by design, and we have no requirement for Microsoft to access them.

However, this alert is raised as a HIGH when actually we are in a more secure state than in we enabled Microsoft Services access. Enabling this access then relies purely on identity for control from any other tenant.

i.e. if we enable this setting, we allow EventGrid access from any tenant to be able to connect to our storage account and there are no longer any network level controls. the only protection is as Managed System Identity (if configured)

### Reproduction Steps

```bash
1. Deploy a storage account
2. Do not enable Microsoft network access
3. Trivy config
...
```

### Target

Filesystem

### Scanner

Misconfiguration

### Debug Output

```bash
n/a
```

### Version

```bash
Version: 0.50.0
Vulnerability DB:
  Version: 2
  UpdatedAt: 2024-11-28 12:17:25.022187652 +0000 UTC
  NextUpdate: 2024-11-29 12:17:25.022187241 +0000 UTC
  DownloadedAt: 2024-11-28 13:46:18.7504028 +0000 UTC
Policy Bundle:
  Digest: sha256:5c4c5f5b566ddbc83888108ba8d6c6f23098cf51a848a8d4f40e2865f18861aa
  DownloadedAt: 2024-12-04 15:10:48.9585057 +0000 UTC
```

### Checklist

- [X] Read [the documentation regarding wrong detection](https://aquasecurity.github.io/trivy/dev/community/contribute/discussion/#false-detection)
- [X] Ran Trivy with `-f json` that shows data sources and confirmed that the security advisory in data sources was correct

**Comment by AlexcFrench** [comment]
I would expect this to be a warning if it was enabled...

**Comment by simar7** [comment]
If the check isn't valid for your case, can you not ignore it? Or is there something else to it.

**Reply by AlexcFrench** [reply]
Hi simar7,

The issue I am identifying is that Trivy's default of allowing Microsoft services is less secure than when these services are not allowed.

If we had a 2 firewall configs:

1) Firewall that blocks all traffic
2) Firewall that allows a lot of traffic through (from known addresses)

Which one is more secure? The Trivy default is number 2.

I would suggest the Trivy default is changed similar to:

| bypass configuration | Trivy Alert Level | 
| ------------ | --------------- |
| no bypass | No Alert |
| Logging |  Low |
| Metrics | Low |
| AzureServices | Medium |

In this configuration:

- There is no alert if no Microsoft Services access is alloweed
- There is a low alert for logging / metrics - an exception could be put in if this is the required configuration, and it this is not an unusual configuration
- A Medium alert is raised when Azure Services access is allowed.

This would notify people via the trivy alert if they have enabled any bypass and they could review / put in an exception if the bypass is acutally required - a good checkpoint.

Enabling AzureServices allows the following services to bypass the storage account firewall:

- Azure Backup
- Azure Site Recovery
- Azure DevTest Labs
- Azure Event Grid
- Azure Event Hubs
- Azure Networking
- Azure Monitor
- Azure SQL Data Warehouse (when registered in the subscription)

The majority of people will not be using these services (monitoring is probably the most common and there is a specific bypass name for that) but Trivy is effecitvely requiring this traffic to be allowed to remove the warning.

The bypass should only be enabled if you use these services.

**Reply by simar7** [reply]
While I agree with what you have said, the recommendation from Azure is to enable the bypass. See the reasoning mentioned in this page https://learn.microsoft.com/en-us/azure/storage/blobs/security-recommendations#networking

Another instance where this approach is followed https://docs.datadoghq.com/security/default_rules/azure-storage-account-trusted-microsoft-services-should-be-enabled-for-storage-account-access/ 

I don't have a good answer at this time, we can keep this discussion open to hear more from the community.

**Comment by AlexcFrench** [comment]
Hi Simar7,

Interesting, thanks for the article.

Yes, if we can leave this open for other input that would be valuable. If this is only my opinion then I'll happily accept the defaults!!!!
